x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,172 @@
+
+#### When user is on the first (0th) batch
+When user is on the first batch i.e. Batch 0, it will be re-fetched with the below strategy.
+
+  ```js
+  {
+    /* Fetch Batch 0 */
+    activePage: 0,
+    querySize: 500 /* batch/sample size */
+  }
+  ```
+
+#### When user has already fetched multiple batches
+Let’s say the user has already fetched 3 batches and currently the user is on Batch 2. Clicking on Refetch will reset the result set to 0th batch:
+
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+### Adding a new column
+Adding a new column should not affect the result set that is currently being displayed in front of the user. We should fetch a complete result set ir-respective of the batch user is on.
+
+#### When user is on the first(0th) batch
+When user is on the first batch i.e. Batch 0, it will be re-fetched with the below strategy.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+#### When user has already fetched multiple batches
+
+When user is on the 4th batch i.e. Batch 3, it will be re-fetched with the below strategy. 
+
+```js
+{
+  /* Fetch Batch 0 */
+  activePage: 0,
+  querySize: 2000 /* cumulative 4 batches ( 500 * 4 ) */
+}
+```
+
+One important thing to notice here is that the querySize is 2000 instead of 500 which means we are cumulatively for all the visible batches. This comes with its own performance considerations but since the maximum allowed resultset is 10K, I think we can work through it.
+
+### Removing a columns
+As far as fetch strategy is concerned, there should not be fetching or re-fetching when a column has been removed. Since we already have the data of the column, we want to retain it as it is.
+
+Only Effect should be on the UI side where the specified column is removed from the table.
+
+### Sort
+Any changes in `Sort` should trigger a completely new fetch of Batch 0 from the start ir-respective of the current batch user is on.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+### Timerange
+Any changes in `Timerange` should trigger a completely new fetch of Batch 0 from the start ir-respective of the current batch user is on.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+### filterQuery
+Any changes in `filterQuery` should trigger a completely new fetch of Batch 0 from the start ir-respective of the current batch user is on.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+### Batch Size
+Any changes in `limit/batchSize/sampleSize` should trigger a completely new fetch of Batch 0 from the start ir-respective of the current batch user is on.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
```

Comment:
all of these could be combined?

Reply:
They could be but I preferred to keep it seperate for ease of discoverability and just for visual audit that we have covered all of the sections. No other reason.

Reply:
works for me!